realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -203,7 +202,12 @@ private void postError(final ObjectServerError error) {
                     handler.post(new Runnable() {
                         @Override
                         public void run() {
-                            callback.onError(error);
+                            try {
```

Comment:
I chatted with @kneth about this yesterday as well, e.g. https://jira.mongodb.org/browse/JAVA-2129 do the same. It is less than ideal, but given that it is impossible for developers to reason about where the exception will finally surface, logging it if it happens seems the least bad solution.